installer_win/omsupply-build.bat

```diff
@@ -0,0 +1,26 @@
+@ECHO ##### Removing previous builds #####
+SET installerWorkspace=C:\Program Files (x86)\Jenkins\jobs\omSupplyMain - installers\workspace\omSupply
+@rmdir "%installerWorkspace%\omSupply Web Server" /s /q
+@rmdir "%installerWorkspace%\omSupply Desktop Server" /s /q
+@rmdir "%installerWorkspace%\build" /s /q
+
+@ECHO ##### Starting omsupply builds #####
+mkdir "%installerWorkspace%\omSupply Web Server"
+xcopy "server\configuration" "%installerWorkspace%\omSupply Web Server\configuration" /e /h /c /i
+
+mkdir "%installerWorkspace%\omSupply Desktop Server"
+mkdir "%installerWorkspace%\omSupply Desktop Server\server"
+xcopy "server\configuration" "%installerWorkspace%\omSupply Desktop Server\server\configuration" /e /h /c /i
+
+copy "server\server\src\omSupply.ico" "build\omSupply.ico"
+xcopy "installer_win" "%installerWorkspace%\build" /e /h /c /i
+copy "version.txt" "%installerWorkspace%\version.txt"
+
+@cd "installer_win"
+start /b /wait omsupply-sqlite-build.bat

```

Comment:
I was wondering if we could speed up the build - one aspect is to remove the duplication, there is multiple `yarn install` and `yarn build` and `cargo clean` which would not be needed if you didn't have the builds split out into separate batch files.

I would also run separate lines and take the risk that, for example, the yarn build fails and the cargo build doesn't embed the client - though actually the cargo build would fail in that case..

Also not sure why `cd installer_win` and then `cd ..` is needed?

```
SET installerWorkspace=C:\Program Files (x86)\Jenkins\jobs\omSupplyMain - installers\workspace\omSupply
cd client && yarn install --force && yarn build
cd server && cargo clean

@ECHO ##### Building omSupply for the desktop #####
cd client && yarn electron:build

@ECHO ##### Copying build artifacts to the installer folder ##### 
xcopy "client\packages\electron\out\open mSupply-win32-x64\**" "%installerWorkspace%\omSupply Desktop Server\" /e /h /c /i

@ECHO ##### Building omsupply for the postgres #####
cd server && cargo build --release --features postgres

@ECHO ##### Copying build artifacts to the installer folder ##### 
copy "server\target\release\remote_server.exe" "%installerWorkspace%\omSupply Web Server\omSupply-postgres.exe" && copy "server\target\release\remote_server.exe" "%installerWorkspace%\omSupply Desktop Server\server\omSupply-postgres.exe"

@ECHO ##### Building omsupply for the sqlite #####
cd "server" && cargo build --release 

@ECHO ##### Copying build artifacts to the installer folder ##### 
copy "server\target\release\remote_server.exe" "%installerWorkspace%\omSupply Web Server\omSupply-sqlite.exe" && copy "server\target\release\remote_server.exe" "%installerWorkspace%\omSupply Desktop Server\server\omSupply-sqlite.exe"
```

Reply:
Yeah it would have been clean and simple if the build scrips could be written in a single bat file and I had tried that as well but the bat file execution didn't seem to be run in sequential manner while I had been testing on windows. If the command which likely to take long time for execution then the bat execution simply doesn't wait for it to be finished instead the execution moves to the next command line which makes the build failed for an empty omsupply exe files for setup factory to make installers.

So had to split each individual build into a separate bat file with `wait` flag to start their execution. I also had to chain the series of commands in each separate bat files so that the command executions don't get broken when first command line executes.

Reply:
oh, that's disappointing. I had a similar problem on jenkins, now that you mention it.
Currently using a bat file on the demo server to build.. but that's windows command line, not jenkins 🙄